### PR TITLE
Added undefined performance hint value

### PR DESCRIPTION
**Pull request by apankratovantonp**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by nkogteva — APPROVED** [review]

**Comment by nkogteva** [comment]
@myshevts Are you planning to fix CPU and GPU in another PR?

**Comment by myshevts** [comment]
> @myshevts Are you planning to fix CPU and GPU in another PR?

My understaning that since this PR treats the "" as the UNDEFINED, this exactly what the CPU and GPU already operate

**Comment by nkogteva** [comment]
> > @myshevts Are you planning to fix CPU and GPU in another PR?
> 
> My understaning that since this PR treats the "" as the UNDEFINED, this exactly what the CPU and GPU already operate, so no additional fix for CPU/GPU is required.

It would be nice to add tests

**Comment by myshevts** [comment]
> It would be nice to add tests

absolutely, @apankratovantonp I guess (general) test is the result of quering the property from any plugin doesn't fail.

**Comment by ilya-lavrenov** [comment]
@apankratovantonp let's add tests as requested

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    auto device_name = CommonTestUtils::DEVICE_TEMPLATE;
+
+    for (auto&& property : core.get_property(device_name, ov::supported_properties)) {
+        if (ov::device::id == property) {
+            std::cout << ov::device::id.name() << " : " << core.get_property(device_name, ov::device::id) << std::endl;
+        } else if (ov::enable_profiling == property) {
+            std::cout << ov::enable_profiling.name() << " : " << core.get_property(device_name, ov::enable_profiling) << std::endl;
+        } else if (ov::hint::performance_mode == property) {
+            std::cout << "Default " << ov::hint::performance_mode.name() << " : " << core.get_property(device_name, ov::hint::performance_mode) << std::endl;
+            core.set_property(device_name, ov::hint::performance_mode(ov::hint::PerformanceMode::UNDEFINED));
```
Comment (myshevts) [inline docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:88]: can we have a loop over all values in the enum (so the test is future-proof if we have more modes)

Comment (myshevts) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:88]: you will need to disable the test for devices that fail

Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:88]: It is TEMPLATE plugin only test

Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:88]: There is no general way to automatically iterate over enum values so two values are enough to prove the workability

**Pull request merged**